openedx/features/course_experience/tests/views/test_course_home.py

```diff
@@ -321,6 +322,11 @@ def test_home_page(
             if expected_enroll_message:
                 self.assertContains(response, 'You must be enrolled in the course to see course content.')
 
+        if enable_unenrolled_access and course_visibility == COURSE_VISIBILITY_PUBLIC:
+            if user_type == CourseUserType.UNENROLLED and self.course.invitation_only:
+                if expected_enroll_message:
+                    self.assertContains(response, 'You must be enrolled in the course to see course content.')
```

Comment:
The message text to be checked is `You should be enrolled in the course to fully access the contents.`
It looks like this is not getting checked since the course's `invitation_only` is set to `False`. Is it possible to set the `invitation_only` to `True` and fetch the response again before this step? If not, you can create a new course like you did earlier with `invitation_only` set to `True` (but no need to add a new ddt data). Note that if  you create a new course, you will need to add a check for `invitation_only` in https://github.com/edx/edx-platform/blob/master/openedx/features/course_experience/tests/views/test_course_home.py#L668 and https://github.com/edx/edx-platform/blob/master/openedx/features/course_experience/tests/views/test_course_home.py#L675.

Reply:
Nice catch. I can creat a private course in the setup and make sure that test uses the reponse from that. In that way I will not need to modify any other tests.

        if enable_unenrolled_access and course_visibility == COURSE_VISIBILITY_PUBLIC:
  1             if user_type == CourseUserType.UNENROLLED and self.private_course.invitation_only:
  2                 if expected_enroll_message:
  3                     self.assertContains(private_response, 'You should be enrolled in the course to fully access the contents.')

Is this ok with you?

Reply:
Yup that should be fine. Make sure you handle the check in the other two lines mentioned in my previous comment as well. Otherwise the test cases will fail.

Reply:
I tested it and the additional tests you  mentioned don't actually fail. It's becaue those tests are not using the private course (self.private_course) but are using the public one (self.course). Anyways, I'm pushing the changes to do a full run of tests.

Reply:
Ah perfect! I missed that. Thanks.